### PR TITLE
Expose the parsed signature in structured format

### PR DESCRIPTION
**Pull request by cshung**
It is my attempt to expose the raw token values implicit in the signature to ILSpy. This is to allow ILSpy to show clickable links when we annotate the disassembly with the import cell signature.

See https://github.com/icsharpcode/ILSpy/pull/1991 for more info on how the raw token values are used there.

### CODE REVIEW

**Review by trylek — APPROVED** [review]
LGTM. I have shared a few nitpicks on the implementation but I haven't noticed anything blocking, they can be easily left for some future cleanup.

File: src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs [hunk 1]
```diff
@@ -12,6 +12,30 @@
 
 namespace ILCompiler.Reflection.ReadyToRun
 {
+    /// <summary>
+    /// This represents all possible signatures that is 
+    /// </summary>
+    public interface ReadyToRunSignature
```
[the remark this answers is not in the document]
Comment (jkotas) [reply src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs:18]: Why is this interface in the first place? Can this be a class ? Do you expect it to have methods in future?

[the remark this answers is not in the document]
Comment (cshung) [reply src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs]: I just wasn't thinking too much about it. I agree that making `ReadyToRunSignature` a class is a better design here.

File: src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs [hunk 2]
```diff
@@ -39,11 +63,11 @@ public static string FormatHandle(MetadataReader metadataReader, Handle handle,
             return formatter.EmitHandleName(handle, namespaceQualified, owningTypeOverride, signaturePrefix);
         }
 
-        public static string FormatSignature(IAssemblyResolver assemblyResolver, ReadyToRunReader r2rReader, int imageOffset)
+        public static string FormatSignature(IAssemblyResolver assemblyResolver, ReadyToRunReader r2rReader, int imageOffset, out ReadyToRunSignature result)
         {
             SignatureDecoder decoder = new SignatureDecoder(assemblyResolver, r2rReader.GetGlobalMetadataReader(), r2rReader, imageOffset);
-            string result = decoder.ReadR2RSignature();
-            return result;
+            string answer = decoder.ReadR2RSignature(out result);
```
[the remark this answers is not in the document]
Comment (cshung) [reply src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs:69]: This sounds like a good suggestion, I have opened https://github.com/dotnet/runtime/issues/36219 to track this work.

**Pull request merged**